dbt_style_guide.md

```diff
@@ -1,237 +1,415 @@
 # dbt Style Guide
 
 ## Model Naming
-Our models (typically) fit into three main categories: staging, marts, base/intermediate. For more detail about why we use this structure, check out [this discourse post](https://discourse.getdbt.com/t/how-we-structure-our-dbt-projects/355). The file and naming structures are as follows:
+
+- All objects should be plural.  
+  Example: `stg_stripe__invoices.sql` vs. `stg_stripe__invoice.sql`
+
+- All objects should have a prefix to indicate their purpose in the flow:
+  - `base_` is uncommon, but indicates a process needed before `stg_` - typically when multiple sources are rarely used independently. The `stg_` model is then used to join or union the sources together before more robust transformations happen.  
+     
+     Example: Creating one table for all cleaned location data.
+     - `base_location__addresses.sql`, `base_location__countries.sql`, and `base_location__states.sql`  would serve to clean the data and be the 1:1 relationship between the source.
+     - `stg_location__locations.sql` would serve to join all location-related data as one entity.
+  - `stg_` is used to clean and standardize data before being fundamentally changed in downstream modeling.
+  - `int_` is used to indicate a step towards creating a final model surfaced to stakeholders.
+  - `fct_` is used to indicate a final data set surfaced to stakeholders, and flags data which is in the form of an immutable event stream.
+  - `dim_` is used to indicate a final data set surfaced to stakeholders, and flags data which is used to describe an entity.  
+  - `seed_` is used to indicate that the data set was built from a CSV file
+  using `dbt seed`.
+
+- All models should use the naming convention `<type/dag_stage>_<source/topic>__<additional_context>`. See [this article](https://docs.getdbt.com/blog/stakeholder-friendly-model-names) for more information.

```

Comment:
Here you call it the "dag_stage" but in the list above it's called "purpose in the flow" - can we connect the two?

Reply:
Done! [Commit](https://github.com/dbt-labs/corp/pull/63/commits/1c973622c93fc2736e3e7b9fbbb2c06ba708a915)